### PR TITLE
Convert Unicode Keywords and Transform Fields to LiteMap

### PR DESCRIPTION
**Pull request by sffc**
Progress on #1109

### CODE REVIEW

**Comment by sffc** [comment]
- @zbraniecki for the changes to `icu_locid`
- @Manishearth for the changes to `litemap`

File: components/locid/src/extensions/transform/fields.rs [hunk 1]
```diff
@@ -107,18 +105,18 @@ impl Fields {
...
     /// let key: Key = "h0".parse()
     ///     .expect("Failed to parse a Key.");
     /// assert!(&fields.contains_key(&key));
     /// ```
-    pub fn contains_key<Q>(&self, key: Q) -> bool
+    pub fn contains_key<Q>(&self, key: &Q) -> bool
     where
-        Q: Borrow<Key>,
+        Key: Borrow<Q>,
+        Q: Ord,
```
Comment (sffc) [inline components/locid/src/extensions/transform/fields.rs:117]: Here and below: I changed this function signature to match the signature of `LiteMap::contains_key`, which in turn appears to match the standard library `HashMap::contains_key`.

File: components/locid/src/extensions/unicode/keywords.rs [hunk 2]
```diff
@@ -49,27 +49,25 @@ impl Keywords {
...
+    /// use icu::locid::Locale;
+    /// use icu::locid::extensions::unicode::Keywords;
     ///
-    /// let key: Key = "ca".parse()
-    ///     .expect("Failed to parse a Key.");
-    /// let value: Value = "buddhist".parse()
-    ///     .expect("Failed to parse a Value.");
-    /// let keywords = Keywords::from_vec_unchecked(vec![(key, value)]);
+    /// let loc1 = Locale::from_bytes(b"und-t-h0-hybrid").unwrap();
+    /// let loc2 = Locale::from_bytes(b"und-u-ca-buddhist").unwrap();
```
Comment (zbraniecki) [inline components/locid/src/extensions/unicode/keywords.rs:64]: `nit`: here and elsewhere, can you use `expect` in place of `unwrap` in examples?

**Review by zbraniecki — APPROVED** [review]

**Review by Manishearth — APPROVED** [review]
litemap lgtm

**Review by Manishearth — APPROVED** [review]

File: components/locid/src/extensions/unicode/keywords.rs (hunk 2 above)
Comment (sffc) [reply components/locid/src/extensions/unicode/keywords.rs:64]: As mentioned in the other PR, we can fix this over the `icu_locid` crate in a follow-up.

We should discuss whether we want this to be a policy for all ICU4X docs. I'm not sure.

**Pull request merged**